### ISSUE TITLE
About the implementation of microbatches

### ISSUE THREAD

**Issue by anirban-nath** [ChrisWaites/pyvacy#9]
The example implementation given here shows that data is broken down into minibatches and then into microbatches of 1 or few samples each. I was wondering instead of calculating loss like: -

```
 for X_microbatch, y_microbatch in microbatch_loader(TensorDataset(X_minibatch, y_minibatch)):
        optimizer.zero_microbatch_grad()
        loss = loss_function(model(X_microbatch), y_microbatch)
        loss.backward()
        optimizer.microbatch_step()
    optimizer.step()
```

what if I did: -

```
 for x_minibatch, y_minibatch in minibatch_loader(train_dataset):
        loss = [] #List of losses from each microbatch
        for microbatch in minibatch
               loss.update(loss_function(model(X_microbatch), y_microbatch))
        
        for loss_mini in loss:        
                  optimizer.zero_microbatch_grad()
                  loss_mini.backward(retain_graph = True)
                  optimizer.microbatch_step()
    optimizer.step()
```
Would this implementation give different results?
